src/ResourceManagement/SqlManagement/Sql.Tests/DatabaseCrudScenarioTests.cs

```diff
@@ -0,0 +1,226 @@
+                    RequestedServiceObjectiveName = "Basic",
+                    Tags = tags,
+                };
+                var db4 = sqlClient.Databases.CreateOrUpdate(resourceGroup.Name, server.Name, dbName, db4Input);
+                Assert.NotNull(db4);
+                SqlManagementTestUtilities.ValidateDatabase(db4Input, db4, dbName);
+
+                // Edition
+                //
+                dbName = SqlManagementTestUtilities.GenerateName(testPrefix);
+                var db5Input = new Microsoft.Azure.Management.Sql.Models.Database()
+                {
+                    Location = server.Location,
+                    Edition = "Basic",
+                    Tags = tags,
+                };
+                var db5 = sqlClient.Databases.CreateOrUpdate(resourceGroup.Name, server.Name, dbName, db5Input);
+                Assert.NotNull(db5);
+                SqlManagementTestUtilities.ValidateDatabase(db5Input, db5, dbName);
+
+                sqlClient.Databases.Delete(resourceGroup.Name, server.Name, db1.Name);
+                sqlClient.Databases.Delete(resourceGroup.Name, server.Name, db2.Name);
+                sqlClient.Databases.Delete(resourceGroup.Name, server.Name, db3.Name);
+                sqlClient.Databases.Delete(resourceGroup.Name, server.Name, db4.Name);
+                sqlClient.Databases.Delete(resourceGroup.Name, server.Name, db5.Name);
+            });
+        }
+
+        [Fact]
+        public void TestUpdateDatabase()
+        {
+            string testPrefix = "sqlcrudtest-";
+            string testName = this.GetType().FullName;
+            SqlManagementTestUtilities.RunTestInNewV12Server(testName, "TestUpdateDatabase", testPrefix, (resClient, sqlClient, resourceGroup, server) =>
+            {
+                Dictionary<string, string> tags = new Dictionary<string, string>()
+                    {
+                        { "tagKey1", "TagValue1" }
+                    };
+
+                string dbName = SqlManagementTestUtilities.GenerateName(testPrefix);
+
+                // Create initial database
+                //
+                var dbInput = new Database()
+                {
+                    Location = server.Location,
+                    Collation = "SQL_Latin1_General_CP1_CI_AS",
+                    Edition = "Basic",
+                    MaxSizeBytes = (2 * 1024L * 1024L * 1024L).ToString(),
+                    RequestedServiceObjectiveName = "Basic",
+                    RequestedServiceObjectiveId = Guid.Parse("dd6d99bb-f193-4ec1-86f2-43d3bccbc49c"),
+                    Tags = tags,
+                };
+                var db1 = sqlClient.Databases.CreateOrUpdate(resourceGroup.Name, server.Name, dbName, dbInput);
+                Assert.NotNull(db1);
+                SqlManagementTestUtilities.ValidateDatabase(dbInput, db1, dbName);
+
+                // Upgrade Edition + SLO Name
+                //
+                var updateEditionAndSloInput = new Database()
```

Comment:
these can all be the same [Fact] since they are upgrading a single database through the test